src/tools/icon-utils.ts

```diff
@@ -0,0 +1,194 @@
+import { TFile, getIcon, normalizePath } from 'obsidian'
+import type OmnisearchPlugin from '../main'
+import {
+  isFileImage,
+  isFilePDF,
+  isFileCanvas,
+  isFileExcalidraw,
+  warnDebug,
+} from './utils'
+
+export interface IconPacks {
+  prefixToIconPack: { [prefix: string]: string }
+  iconsPath: string
+}
+
+export async function loadIconData(plugin: OmnisearchPlugin): Promise<any> {
+  const app = plugin.app
+
+  // Check if the 'obsidian-icon-folder' plugin is installed and enabled
+  // Casting 'app' to 'any' here to avoid TypeScript errors since 'plugins' might not be defined on 'App'
+  const iconFolderPlugin = (app as any).plugins.getPlugin(
+    'obsidian-icon-folder'
+  )
+  if (!iconFolderPlugin) {
+    warnDebug('Icon Folder plugin is not installed or not enabled.')
```

Comment:
Just remove this warn - like I said, it's not unexpected to _not_ have the plugin so no need to have a log